### PR TITLE
feat: add star wars api

### PR DESCRIPTION
**Pull request by favna**
| ⚠️ This Pull Request needs to be rebase & merged. |
| --- |

TODO:

- [x] Remove `old-stuff-TODO-remove/old-star-wars-command.ts`
- [x] Write `swplanet.ts`
- [x] Write `swspecies.ts`
- [x] Write `swstarship.ts`
- [x] Write `swvehicle.ts`
- [x] Validate embed layout with peers

### CODE REVIEW

File: typescript/src/languages/en-US/commands/starwars.json [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+			"hope",
+            "revenge of the sith"
+		],
+        "reminder": "At the moment only data on the first 6 movies is available. Data in movies 7 through 9 and spin-offs might be added in the future."
+    },
+    "filmQueryFail": "{{REDCROSS}} I am sorry, but that query failed. Are you sure `{{film}}` is actually a Star Wars movie?",
+    "filmEmbedTitles": {
+        "characters": "People in this film",
+        "creationDate": "Creation date",
+        "director": "Directory",
```
Comment (thehairy) [inline typescript/src/languages/en-US/commands/starwars.json]:
```suggestion
        "director": "Director",
```

File: typescript/src/languages/en-US/commands/tools.json [hunk 2]
```diff
@@ -587,5 +587,89 @@
...
+			"starships Millenium Falcon",
+			"vehicles Snowspeeder"
+		]
+	},
+	"starWarsNoResult": "{{REDCROSS}} I'm sorry, but I could not find anything for {{query}}",
+	"starWarsPeopleTitles": {
+		"films": {
+			"characters": "Top 3 characters in this film",
+			"creationDate": "Creation date",
+			"director": "Directory",
```
Comment (thehairy) [inline typescript/src/languages/en-US/commands/tools.json]:
```suggestion
			"director": "Director",
```

**Review by thehairy — CHANGES_REQUESTED** [review]
I do think "Directory" has a different meaning. xD

**Review by thehairy — DISMISSED** [review]
Now dis beautiful. 😄

File: typescript/src/languages/en-US/commands/starwars.json [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        "vehicles": "Vehicles in this film"
+    },
+    "peopleDescription": "Get data for any given character in Star Wars",
+    "peopleExtended": {
+        "usages": [
+            "Person"
+        ],
+        "extendedHelp": "Uses a fuzzy search on the name of the person",
+        "explainedUsage": [
+			[
```
Comment (kyranet) [inline typescript/src/languages/en-US/commands/starwars.json]: This file is a soup of spaces vs tabs.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: typescript/src/languages/en-US/commands/starwars.json (hunk 3 above)
Comment (favna) [reply typescript/src/languages/en-US/commands/starwars.json]: Fucking hell I blame that I removed those options from vscode settings that I re-added in this PR.

File: typescript/src/commands/Starwars/swfilm.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+				.setThumbnail(CdnUrls.StarWarsLogo)
+		});
+
+		const filmTitles = t(LanguageKeys.Commands.StarWars.FilmEmbedTitles);
+
+		for (const result of results) {
+			display
+				.addPageEmbed((embed) => {
+					const description = [
+						`**${filmTitles.episodeId}**: ${result.episodeId}`, //
```
Comment (Tylertron1998) [inline typescript/src/commands/Starwars/swfilm.ts]: Why are there random `//` (empty comments) here?

File: typescript/src/commands/Starwars/swfilm.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+						.setTitle(result.title) //
+						.setDescription(description.join('\n\n'));
+				});
+		}
+
+		await display.run(loadingMessage, message.author);
+		return loadingMessage;
+	}
+
+	private async fetchAPI(film: string) {
```
Comment (Tylertron1998) [inline typescript/src/commands/Starwars/swfilm.ts:108]: not sure if it's possible- but if so, might be a good idea to make a `StarWarsPaginatedCommand` class, and have the `fetchAPI` as a generic base method on the class, and inherit from that? `fetchAPI` seems to be pretty much duplicated with different type methods everywhere. If not possible then all good, though.

File: typescript/src/languages/en-US/commands/starwars.json [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+			]
+		],
+        "examples": [
+			"1",
+			"hope",
+            "revenge of the sith"
+		],
+        "reminder": "At the moment only data on the first 6 movies is available. Data in movies 7 through 9 and spin-offs might be added in the future."
+    },
+    "filmQueryFail": "{{REDCROSS}} I am sorry, but that query failed. Are you sure `{{film}}` is actually a Star Wars movie?",
```
Comment (Tylertron1998) [inline typescript/src/languages/en-US/commands/starwars.json]: might be a good shout to remind that only 1-6 movies are available via the API, at the moment.

**Review by Tylertron1998 — COMMENTED** [review]
other then a few questions, looks good to me. Great job!

File: typescript/src/commands/Starwars/swfilm.ts (hunk 4 above)
Comment (Favna) [reply typescript/src/commands/Starwars/swfilm.ts:40]: Ensuring Prettier doesn't put everything on 1 line

File: typescript/src/commands/Starwars/swfilm.ts (hunk 5 above)
Comment (favna) [reply typescript/src/commands/Starwars/swfilm.ts:108]: Not worth it because of variables tbh

**Review by Tylertron1998 — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**